src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CollectionBackedStoreProvider.cs

```diff
@@ -0,0 +1,159 @@
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal class CollectionBackedStoreProvider : IStorePal
+    {
+        private readonly X509Certificate2Collection _certs;
+
+        internal CollectionBackedStoreProvider(X509Certificate2 cert)
+        {
+            _certs = new X509Certificate2Collection(cert);
+        }
+
+        internal CollectionBackedStoreProvider(X509Certificate2Collection certs)
+        {
+            _certs = new X509Certificate2Collection(certs);
+        }
+
+        public void Dispose()
+        {
+        }
+
+        public IEnumerable<X509Certificate2> Find(X509FindType findType, object findValue, bool validOnly)
+        {
+            return Array.Empty<X509Certificate2>();
```

Comment:
I think this is just telling me that I need to reconstruct the PAL interface here a bit.  In Windows the export operations happen on the same data structure as the X509Store; so the export code says "make a memory store, then call export on it"; and this is an implementation of that memory store just for the duration of a call to Export; so calls to Find shouldn't happen (which means it should probably be an Exception; but I copy/pasted this from "Find is valid, but doesn't work yet" in the system stores).

But, that said, I'd prefer to do that reshaping to a "Here, export this collection" instead of "Make a store from a collection; hey store, go export yourself" as a followup, so I can try to get the changes that depend on this (black box) in for beta7 :).